go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -343,7 +361,12 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 			return bsnc.removeRemoteZonePodFromNamespaceAddressSet(pod)
 		}
 
-		return nil
+		// except for localnet networks, continue the delete flow in case a node just
+		// became remote where we might still need to cleanup. On L3 networks
+		// the node switch is removed so there is no need to do this.
+		if bsnc.TopologyType() != types.LocalnetTopology {
```

Comment:
wait does this mean on every remote pod delete we will try to delete locally? That seems like a performance issue with executing txns for no reason? Shouldn't we at least check if the port actually exists before we try to delete?

Reply:
Internally model client does exactly that, it checks if the port exists in the cache before sending any transaction towards the DB. If it is not in the cache it won't send any transaction to the DB.